pkg/v1/tkr/pkg/registry/client.go

```diff
@@ -77,7 +77,7 @@ func getFileContentFromImage(image regv1.Image, filename string) ([]byte, error)
 			return nil, err
 		}
 
-		defer layerStream.Close()
+		defer layerStream.Close() //nolint:gocritic
```

Comment:
question: why do we want to `nolint:gocritic` here ? If there is really something that is "won't fix", could you also add some comments to provide some details ?

Reply:
i did not want to fix something that is not part of this pr and break things, also, it's not our code, but the linter was complaining.. so i just added `nolint:gocricitc` to suppress the linter. I was planning to let the tkr team know about this so that they can make appropriate fix

Reply:
add comment and created a gh issue to track this https://github.com/vmware-tanzu/tanzu-framework/issues/3917